public/en/platform/protect-php-sdk/configuration-documentation.md

```diff
@@ -0,0 +1,81 @@
+## Configuration Manager
+
+### Purpose of The Configuration Manager
+The Config Manager class is intended to be a flexible, functional class that can be used as a static class to manage configurable values during runtime or as a class instance to be used repeatedly throughout a method. Given this, the developmental goals of implementing the class were:
+
+  * Avoid excessive initialization of static configurable values (e.g. avoid loading config files or fetching config data multiple times)
+
+  * Provide a simple service that allows developers to set, check, and retrieve configurable values such as version info, environment relevant URIs, & base values required for authentication.
+
+  * Provide a simple structure for fetching configuration data and allow configuration data to be structured in a way that is intuitive to add, read, & edit.
+
+
+### Example Configuration Manager Usages
+The Config Manager is explicitly designed for abstract usage. The following implementations are example usages of the Config Manager:
+
+JSON Sample For Examples
+```json
+{
+  "version": 2,
+  "platform": "Magento",
+  "foo": "bar",
+  "log_http_calls": true,
+  "production": {
+    "api_url": "https://ns8.com",
+    "logging_path": "logs/production_log.log"
+  },
+  "testing": {
+    "api_url": "https://testing.ns8.com",
+    "logging_path": "logs/testing_log.log"
+  },
```

Comment:
I think for these two cases, you should not be allowed to change the URLs.

Reply:
@crfroehlich I'll add a check function in the `setValue` method that prohibits overriding them in the PHP code.